### PR TITLE
issue-52: an invalid date in the date fields blocks everything

### LINKED ISSUE #52: an invalid date in the date fields blocks everything
image

### PR DESCRIPTION
**Pull request by giorgio94c**
Close #52

### CODE REVIEW

File: public/kibi_timeline.js [hunk 1]
```diff
@@ -164,7 +164,8 @@ define(function (require) {
             var endRawFieldValue;
 
             _.each(searchResp.hits.hits, function (hit) {
-              if (hit.fields[params.startField] === null || hit.fields[params.startField] === undefined) {
+              if (hit.fields[params.startField] === null || hit.fields[params.startField] === undefined
```
Comment (szydan) [inline public/kibi_timeline.js]: Why we need this check here ?
In line 178 we are already getting the startValue and there is elready **if** there
Why we can NOT just add else to it ??
And why we have to test hit.fields against all 3 different values like null, undefined and NaN
I think the  getDescendantPropValue should return null if there is no value ?
Please ad a test for this case (there are already tests for this method) and then if it is true please remove this unnecessary checks and just print warning in else block

File: public/kibi_timeline.js [hunk 2]
```diff
@@ -238,6 +230,12 @@ define(function (require) {
                   }
                 }
                 events.push(e);
+              } else {
+                if ($scope.notifyNull) {
```
Comment (szydan) [inline public/kibi_timeline.js]: ok better now :-)
Cold you please change the name of the flag to notifyDataErrors as it should trigger if the value is either 
undefined null or empty string

Make sure that if the value is an empty string this is also triggered

File: public/lib/helpers/__tests__/timeline_helper.js [hunk 3]
```diff
@@ -4,6 +4,11 @@ var timelineHelper = require('../timeline_helper')();
 describe('TimelineHelper', function () {
 
   describe('getDescendantPropValue', function () {
+    it('no data', function () {
+      var o = {};
+      expect(timelineHelper.getDescendantPropValue(o, 'p1.p2')).to.be(undefined);
+    });
+
```
Comment (szydan) [inline public/lib/helpers/__tests__/timeline_helper.js]: please add another test for null and empty string 
They both should return respectively null and empty string

File: public/kibi_timeline.js [hunk 4]
```diff
@@ -238,6 +230,12 @@ define(function (require) {
                   }
                 }
                 events.push(e);
+              } else {
+                if ($scope.notifyDataErrors || $scope.notifyDataErrors === '') {
```
Comment (szydan) [inline public/kibi_timeline.js]: Instead of checking here if it is  === ''
It would be better to set the default value to false here 
https://github.com/sirensolutions/kibi_timeline_vis/blob/master/public/kibi_timeline_vis.js#L22
